src/charm.py

```diff
@@ -7,32 +7,43 @@
 
 import logging
 from contextlib import contextmanager
-from typing import Dict, List, Optional
+from typing import Dict, List, Optional, Tuple
 
 import hcl
 from charms.grafana_agent.v0.cos_agent import COSAgentProvider
 from charms.operator_libs_linux.v2 import snap
+from charms.vault_k8s.v0.vault_client import (
+    AppRole,
+    AuditDeviceType,
+    Token,
+    Vault,
+    VaultClientError,
+)
 from charms.vault_k8s.v0.vault_tls import (
     File,
     VaultTLSManager,
 )
 from jinja2 import Environment, FileSystemLoader
 from machine import Machine
+from ops import ActionEvent, BlockedStatus, Secret, SecretNotFoundError
 from ops.charm import CharmBase, CollectStatusEvent
 from ops.main import main
 from ops.model import ActiveStatus, MaintenanceStatus, ModelError, WaitingStatus
 
 logger = logging.getLogger(__name__)
 
+CHARM_POLICY_NAME = "charm-access"
+CHARM_POLICY_PATH = "src/templates/charm_policy.hcl"
 CONFIG_TEMPLATE_DIR_PATH = "src/templates/"
 CONFIG_TEMPLATE_NAME = "vault.hcl.j2"
 PEER_RELATION_NAME = "vault-peers"
+VAULT_CHARM_APPROLE_SECRET_LABEL = "vault-approle-auth-details"
 VAULT_CONFIG_PATH = "/var/snap/vault/common"
 VAULT_CONFIG_FILE_NAME = "vault.hcl"
+VAULT_DEFAULT_POLICY_NAME = "default"
```

Comment:
What is this policy? I only see us creating the charm policy, not this default one.

Reply:
https://developer.hashicorp.com/vault/docs/concepts/policies#default-policy